lib/perl/Genome/InstrumentData/AlignmentResult.pm

```diff
@@ -1963,5 +1963,61 @@ sub add_to_temporary_input_files_queue {
     return 1;
 }
 
+sub lock_bam_file_access {
+    my $self = shift;
+
+    unless ($self->get_merged_alignment_results) {
+        my @bams = $self->alignment_bam_file_paths;
+        unless (@bams) {
+            die $self->error_message("Alignment result with class (%s) and id (%s) has neither ".
```

Comment:
alignment_bam_file_paths is used here only to check for some extreme cases where neither merged alignment results nor per lan bams exist. If the merge from other process commits, removes the per lane bam and releases the lock, this process will get the lock, but immediately unlock this lock after checking get_merged_alignment_results again that returns true this time.